src/plugins/vis_type_vega/public/expressions/line_vega_spec_fn.ts

```diff
@@ -0,0 +1,243 @@
+    const xAxisId = getXAxisId(dimensions, datatable.columns);
+    const xAxisTitle = cleanString(dimensions.x.label);
+    const startTime = new Date(dimensions.x.params.bounds.min).valueOf();
+    const endTime = new Date(dimensions.x.params.bounds.max).valueOf();
+    let skip = 0;
+    datatable.columns.forEach((column, index) => {
+      // Check if its not xAxis column data
+      if (column.meta?.aggConfigParams?.interval != null) {
+        skip++;
+      } else {
+        const currentSeriesParams = visParams.seriesParams[index - skip];
+        const currentValueAxis =
+          // @ts-ignore
+          valueAxis[currentSeriesParams.valueAxis.toString()];
+        let tooltip: Array<{ field: string; type: string; title: string }> = [];
+        if (visParams.addTooltip) {
+          tooltip = [
+            { field: xAxisId, type: 'temporal', title: xAxisTitle },
+            { field: column.id, type: 'quantitative', title: column.name },
+          ];
+        }
+        spec.layer.push({
+          mark: {
+            // Possible types are: line, area, histogram. The eligibility checker will
+            // prevent area and histogram (though area works in vega-lite)
+            type: currentSeriesParams.type,
+            // Possible types: linear, cardinal, step-after. All of these types work in vega-lite
+            interpolate: currentSeriesParams.interpolate,
+            // The possible values is any number, which matches what vega-lite supports
+            strokeWidth: currentSeriesParams.lineWidth,
+            // this corresponds to showing the dots in the visbuilder for each data point
+            point: currentSeriesParams.showCircles,
+          },
+          encoding: {
+            x: {
+              axis: {
+                title: xAxisTitle,
+                grid: visParams.grid.categoryLines,
+              },
+              field: xAxisId,
+              type: 'temporal',
+              scale: {
+                domain: [startTime, endTime],
+              },
+            },
+            y: {
+              axis: {
+                title: cleanString(currentValueAxis.title.text) || column.name,
+                grid: visParams.grid.valueAxis !== '',
+                orient: currentValueAxis.position,
+                labels: currentValueAxis.labels.show,
+                labelAngle: currentValueAxis.labels.rotate,
+              },
+              field: column.id,
+              type: 'quantitative',
+            },
+            tooltip,
+            color: {
+              datum: column.name,
+            },
+          },
+        });
+      }
+    });
+  }
+
+  if (visParams.addTimeMarker) {
+    spec.transform = [
+      {
+        calculate: 'now()',
+        as: 'now_field',
+      },
+    ];
+
+    spec.layer.push({
+      mark: 'rule',
+      encoding: {
+        x: {
+          type: 'temporal',
+          field: 'now_field',
+        },
+        color: {
+          value: 'red',
+        },
```

Comment:
for time marker, what is the existing color look like? if it's red, i'm ok. If it's not, i prefer we have some neutral color and reserve red for the events.

I'll also confirm if the event rules should be red or something else.

Reply:
It is red, so we are keeping it consistent